app/validators/immigration_status_validator.rb

```diff
@@ -0,0 +1,25 @@
+class ImmigrationStatusValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, application_choice)
+    return if did_not_add_nationality_yet?(application_choice) || immigration_right_to_work?(application_choice) || course_can_sponsor_visa?(application_choice)
+
+    record.errors.add(attribute, :immigration_status)
+  end
+
+  def did_not_add_nationality_yet?(application_choice)
+    application_choice.application_form.nationalities.blank?
+  end
+
+  def immigration_right_to_work?(application_choice)
+    application_choice.application_form.british_or_irish? ||
+      application_choice.application_form.right_to_work_or_study_yes?
+  end
+
+  def course_can_sponsor_visa?(application_choice)
```

Comment:
Did you copy these two from `CourseChoicesReviewComponent`? Is it worth centralising it somewhere to avoid duplication? Easiest I can think of would be the model.

Reply:
CourseChoicesReviewComponent is not used anymore. When we clean up the non continuous application the whole code will be deleted. It not even running this code at the moment. I think I will leave as it is.